core/src/test/scala/org/apache/spark/scheduler/MapStatusSuite.scala

```diff
@@ -128,4 +130,23 @@ class MapStatusSuite extends SparkFunSuite {
     assert(size1 === size2)
     assert(!success)
   }
+
+  test("Blocks which are bigger than 2 * average size should not be underestimated.") {
+    val sizes = Array.concat(Array.fill[Long](1000)(1L), (1000L to 2000L).toArray)
+    val status1 = MapStatus(BlockManagerId("exec-0", "host-0", 100), sizes)
+    val arrayStream = new ByteArrayOutputStream(102400)
+    val objectOutputStream = new ObjectOutputStream(arrayStream)
+    assert(status1.isInstanceOf[HighlyCompressedMapStatus])
+    status1.asInstanceOf[HighlyCompressedMapStatus].writeExternal(objectOutputStream)
```

Comment:
I see the problem. This is not the normal process for serializing java objects, `writeExternal`/`readExternal` should be called by java serialization framework, instead of being called manually. We should do `objectOutputStream.writeObject(status1)`